File: temp_analyze_pdf.js
const fs = require('fs');
const https = require('https');
const { PDFDocument } = require('pdf-lib');

async function analyze() {
    const url = "https://www.rd.go.th/fileadmin/tax_pdf/withhold/approve_wh3_081156.pdf";
    console.log("Downloading PDF...");
    
    const file = fs.createWriteStream("temp_doc.pdf");
    https.get(url, function(response) {
        response.pipe(file);
        file.on('finish', async function() {
            file.close();
            console.log("Download complete. Analyzing...");
            try {
                const pdfBytes = fs.readFileSync("temp_doc.pdf");
                const pdfDoc = await PDFDocument.load(pdfBytes);
                
                console.log(`Number of Pages: ${pdfDoc.getPageCount()}`);
                console.log(`Title: ${pdfDoc.getTitle()}`);
                console.log(`Author: ${pdfDoc.getAuthor()}`);
                console.log(`Creation Date: ${pdfDoc.getCreationDate()}`);
                console.log(`Modification Date: ${pdfDoc.getModificationDate()}`);
                
                const form = pdfDoc.getForm();
                const fields = form.getFields();
                console.log(`\nForm Fields found: ${fields.length}`);
                
                if (fields.length > 0) {
                    console.log("First 30 fields:");
                    fields.slice(0, 30).forEach(field => {
                        console.log(`- ${field.getName()} (${field.constructor.name})`);
                    });
                } else {
                    console.log("No interactive form fields found.");
                }
            } catch (err) {
                console.error("Error analyzing PDF:", err);
            }
        });
    }).on('error', function(err) {
        console.error("Download error:", err);
    });
}

analyze();

File: temp_analyze_pdf.py
import urllib.request
import os
import sys

def analyze():
    url = "https://www.rd.go.th/fileadmin/tax_pdf/withhold/approve_wh3_081156.pdf"
    print("Downloading PDF...")
    try:
        filename, _ = urllib.request.urlretrieve(url, "temp_doc.pdf")
    except Exception as e:
        print(f"Error downloading: {e}")
        return
        
    try:
        from pypdf import PdfReader
    except ImportError:
        print("Installing pypdf...")
        os.system(f"{sys.executable} -m pip install pypdf")
        from pypdf import PdfReader
        
    try:
        reader = PdfReader("temp_doc.pdf")
        print(f"Number of Pages: {len(reader.pages)}")
        print(f"Metadata: {reader.metadata}")
        
        fields = reader.get_fields()
        if fields:
            print(f"\nForm Fields found: {len(fields)}")
            for key in list(fields.keys())[:30]:
                print(f"- {key}")
        else:
            print("\nNo interactive form fields found.")
            
        print("\n--- First Page Text Preview ---")
        text = reader.pages[0].extract_text()
        print(text[:1000] if text else "No text extracted.")
        
    except Exception as e:
        print(f"Error reading PDF: {e}")

if __name__ == '__main__':
    analyze()
